openedx/core/djangoapps/content/course_views/tabs.py

```diff
@@ -0,0 +1,573 @@
+
+    def __getitem__(self, key):
+        if key == 'is_hidden':
+            return self.is_hidden
+        else:
+            return super(HideableTab, self).__getitem__(key)
+
+    def __setitem__(self, key, value):
+        if key == 'is_hidden':
+            self.is_hidden = value
+        else:
+            super(HideableTab, self).__setitem__(key, value)
+
+    def to_json(self):
+        to_json_val = super(HideableTab, self).to_json()
+        if self.is_hidden:
+            to_json_val.update({'is_hidden': True})
+        return to_json_val
+
+    def __eq__(self, other):
+        if not super(HideableTab, self).__eq__(other):
+            return False
+        return self.is_hidden == other.get('is_hidden', False)
+
+
+class CoursewareTab(EnrolledOrStaffTab):
+    """
+    A tab containing the course content.
+    """
+
+    type = 'courseware'
+    name = 'courseware'
+    is_movable = False
+
+    def __init__(self, tab_dict=None):  # pylint: disable=unused-argument
+        super(CoursewareTab, self).__init__(
+            # Translators: 'Courseware' refers to the tab in the courseware that leads to the content of a course
+            name=_('Courseware'),  # support fixed name for the courseware tab
+            tab_id=self.type,
+            link_func=link_reverse_func(self.type),
+        )
+
+
+class CourseInfoTab(CourseTab):
+    """
+    A tab containing information about the course.
+    """
+
+    type = 'course_info'
+    name = 'course_info'
+    is_movable = False
+
+    def __init__(self, tab_dict=None):
+        super(CourseInfoTab, self).__init__(
+            # Translators: "Course Info" is the name of the course's information and updates page
+            name=tab_dict['name'] if tab_dict else _('Course Info'),
+            tab_id='info',
+            link_func=link_reverse_func('info'),
+        )
+
+    @classmethod
+    def validate(cls, tab_dict, raise_error=True):
+        return super(CourseInfoTab, cls).validate(tab_dict, raise_error) and need_name(tab_dict, raise_error)
+
+
+class ProgressTab(EnrolledOrStaffTab):
+    """
+    A tab containing information about the authenticated user's progress.
+    """
+
+    type = 'progress'
+    name = 'progress'
+
+    def __init__(self, tab_dict=None):
+        super(ProgressTab, self).__init__(
+            # Translators: "Progress" is the name of the student's course progress page
+            name=tab_dict['name'] if tab_dict else _('Progress'),
+            tab_id=self.type,
+            link_func=link_reverse_func(self.type),
+        )
+
+    def is_enabled(self, course, settings, user=None):
+        super_can_display = super(ProgressTab, self).is_enabled(course, settings, user=user)
+        return super_can_display and not course.hide_progress_tab
+
+    @classmethod
+    def validate(cls, tab_dict, raise_error=True):
+        return super(ProgressTab, cls).validate(tab_dict, raise_error) and need_name(tab_dict, raise_error)
+
+
+class WikiTab(HideableTab):
```

Comment:
I was thinking that you'd convert these over to plugins. It would be great to have no special knowledge of the wiki tab anywhere, especially in ``common/lib/xmodule``. It would be great if the wiki were truly optional and that some Open edX installations could run without it, or with a different solution.